sdk/storage/azure-storage-quickquery/src/main/java/com/azure/storage/quickquery/implementation/util/FluxInputStream.java

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.quickquery.implementation.util;
+
+import com.azure.core.util.FluxUtil;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.storage.blob.models.BlobStorageException;
+import com.azure.storage.common.implementation.Constants;
+import org.reactivestreams.Subscription;
+import reactor.core.publisher.Flux;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * An InputStream interface that subscribes to a Flux allows data to be read.
+ */
+public class FluxInputStream extends InputStream {
+
+    private ClientLogger logger;
+
+    // The data to subscribe to.
+    private Flux<ByteBuffer> data;
+
+    // Subscription to request more data from as needed
+    private Subscription subscription;
+
+    private ByteArrayInputStream buffer;
+
+    private boolean subscribed;
+    private boolean fluxComplete;
+    private boolean waitingForData;
+
+    /* The following lock and condition variable is to synchronize access between the reader and the
+        reactor thread asynchronously reading data from the Flux. If no data is available, the reader
+        acquires the lock and waits on the dataAvailable condition variable. Once data is available
+        (or an error or completion event occurs) the reactor thread acquires the lock and signals that
+        data is available. */
+    private final Lock lock;
+    private final Condition dataAvailable;
+
+    private IOException lastError;
+
+    /**
+     * Creates a new FluxInputStream
+     *
+     * @param data The data to subscribe to and read from.
+     * @param logger {@link ClientLogger}
+     */
+    public FluxInputStream(Flux<ByteBuffer> data, ClientLogger logger) {
```

Comment:
I think FluxInputStream should have its own client logger rather than being passed one